From ad1c45e8151be41a319788679daa192bceea3923 Mon Sep 17 00:00:00 2001
From: Deokgyu Yang <secugyu@gmail.com>
Date: Thu, 15 Apr 2021 18:25:05 +0900
Subject: [PATCH 82/99] ODROID-C4: arm64/dts: Make UART_A disabled by default
 and add UART DTBOs

Since UART_A can be enabled by DTBO now.

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: Ie08f49bafc7fab96879469deeaa53b5d7c9a9dfa
---
 .../boot/dts/amlogic/overlays/odroidc4/Makefile     |  4 ++++
 .../boot/dts/amlogic/overlays/odroidc4/uart0.dts    | 13 +++++++++++++
 .../boot/dts/amlogic/overlays/odroidc4/uart1.dts    | 13 +++++++++++++
 .../boot/dts/amlogic/overlays/odroidc4/uart2.dts    | 13 +++++++++++++
 .../boot/dts/amlogic/overlays/odroidc4/uart3.dts    | 13 +++++++++++++
 5 files changed, 56 insertions(+)
 create mode 100644 arch/arm64/boot/dts/amlogic/overlays/odroidc4/uart0.dts
 create mode 100644 arch/arm64/boot/dts/amlogic/overlays/odroidc4/uart1.dts
 create mode 100644 arch/arm64/boot/dts/amlogic/overlays/odroidc4/uart2.dts
 create mode 100644 arch/arm64/boot/dts/amlogic/overlays/odroidc4/uart3.dts

diff --git a/arch/arm64/boot/dts/amlogic/overlays/odroidc4/Makefile b/arch/arm64/boot/dts/amlogic/overlays/odroidc4/Makefile
index ed4aa42b6722..02864e3402ba 100644
--- a/arch/arm64/boot/dts/amlogic/overlays/odroidc4/Makefile
+++ b/arch/arm64/boot/dts/amlogic/overlays/odroidc4/Makefile
@@ -4,6 +4,10 @@ dtbo-y += \
 	pcf8563.dtbo \
 	spi0.dtbo \
 	sx865x-i2c1.dtbo \
+	uart0.dtbo \
+	uart1.dtbo \
+	uart2.dtbo \
+	uart3.dtbo \
 	w1-gpio_p15.dtbo \
 	w1-gpio_p22.dtbo \
 	hifishield2.dtbo
diff --git a/arch/arm64/boot/dts/amlogic/overlays/odroidc4/uart0.dts b/arch/arm64/boot/dts/amlogic/overlays/odroidc4/uart0.dts
new file mode 100644
index 000000000000..a9cb3fb41077
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/overlays/odroidc4/uart0.dts
@@ -0,0 +1,13 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+	fragment@0 {
+		target = <&uart_A>;
+
+		__overlay__ {
+			status = "okay";
+		};
+	};
+};
+
diff --git a/arch/arm64/boot/dts/amlogic/overlays/odroidc4/uart1.dts b/arch/arm64/boot/dts/amlogic/overlays/odroidc4/uart1.dts
new file mode 100644
index 000000000000..c1d25f9bee4c
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/overlays/odroidc4/uart1.dts
@@ -0,0 +1,13 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+	fragment@0 {
+		target = <&uart_B>;
+
+		__overlay__ {
+			status = "okay";
+		};
+	};
+};
+
diff --git a/arch/arm64/boot/dts/amlogic/overlays/odroidc4/uart2.dts b/arch/arm64/boot/dts/amlogic/overlays/odroidc4/uart2.dts
new file mode 100644
index 000000000000..e2d717288ed0
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/overlays/odroidc4/uart2.dts
@@ -0,0 +1,13 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+	fragment@0 {
+		target = <&uart_C>;
+
+		__overlay__ {
+			status = "okay";
+		};
+	};
+};
+
diff --git a/arch/arm64/boot/dts/amlogic/overlays/odroidc4/uart3.dts b/arch/arm64/boot/dts/amlogic/overlays/odroidc4/uart3.dts
new file mode 100644
index 000000000000..735faedc9d94
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/overlays/odroidc4/uart3.dts
@@ -0,0 +1,13 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+	fragment@0 {
+		target = <&uart_AO_B>;
+
+		__overlay__ {
+			status = "okay";
+		};
+	};
+};
+
-- 
2.37.3

